### ISSUE TITLE
[Accessibility] Default theme: form fields not associated to their label make them impossible to fill for blind users

### ISSUE THREAD

**Issue by juliemoynat** [PrestaShop/PrestaShop#23444]
#### Describe the bug

A lot of forms in the default Prestashop theme have form fields not associated to their label so it's making them impossible to fill for blind users. **It's a huge accessibility problem.**

As a result, a lot of Prestashop shops do not have accessible forms because the default theme is a model for developers…

#### Expected behavior

Associate form fields to their label. It's a basic HTML feature where you need to have a `for` attribute on `label` elements and an `id` attribute on form fields (`input`, `textarea`, `select`). The value of the `for` attribute must be the value of its field `id`. IDs must be unique in a page, of course, so that it can work.

Example:

```html
<label for="username">Username</label>
<input id="username" type="text" name="username">
```

[See the WAI doc here](https://www.w3.org/WAI/tips/developing/#associate-a-label-with-every-form-control)

#### Steps to Reproduce

1. Go on the demo site: https://demo.prestashop.com/#/en/front
2. Go on create account page or sign in, or add products to cart and go to checkout: guest checkout or sign in forms, etc.
3. Open the browser development tools and see that there is no `for` attribute on `label` elements nor `id` attribute on form fields. A screen reader, tool used by blind people to navigate in their computer, will not be able to tell what value the field is expecting.

#### Additional information

* PrestaShop version: N/A
* PHP version: N/A

**Comment by matks** [comment]
Hi @juliemoynat in PR https://github.com/PrestaShop/PrestaShop/pull/23353/ (which has been merged in `develop` so you cannot see this on demo.prestashop.com), we use `aria-label`. Is that a suitable replacement for the use of `for` and `id` ?

**Comment by jeckyl** [comment]
> Hi @juliemoynat in PR #23353 (which has been merged in `develop` so you cannot see this on demo.prestashop.com), we use `aria-label`. Is that a suitable replacement for the use of `for` and `id` ?

https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-label
https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-labelledby

Thanks @juliemoynat with your feedback regarding this crucial point in modern development.

personally, I never got used to using the for="" option because I always included my input in the label in order to save resources by reducing the source code, certainly a remnant of the modem era at 33kbauds.

**Comment by juliemoynat** [comment]
Hi @matks, I've read the PR and there is only two fields concerned by it if I read correctly so we can't consider that this PR fix my ticket. ;-)

I do not recommend the use of `aria-label` for several reasons:

- The first rule of ARIA is ["No ARIA is better than bad ARIA"](https://www.w3.org/TR/2019/WD-wai-aria-practices-1.2-20191218/#no_aria_better_bad_aria). If there is an HTML element for that, use HTML.
- Only a few people really know how to well use ARIA and it's a real risk to use it in a theme that serves as a model because you can be sure it will be used by people who would not know nothing about it.
- `aria-label` is an attribute which replace all other names of an element by the `aria-label` content. For example in this code `<label for="first-name">First name</label><input id="first-name" type="text" aria-label="Name" />`, the name of the field will be "Name" and not "First name". So, a user using voice control will tell "Go in First name field" and the field will not be found because it's name is "Name", not "First name" because of the `aria-label`. That's dangerous.
- The `aria-label` attribute is not translated by automatic translation by browsers ([source](https://adrianroselli.com/2019/11/aria-label-does-not-translate.html)).

**Using the `for` and `id` combination for `label` and form fields is the more robust way to have accessible form fields** (and in most of the forms in the Prestashop theme, `label`s already exist so there is absolutely no reason to use `aria-label` instead of `for` and `id` combination).

**Comment by hibatallahAouadni** [comment]
Hello @juliemoynat 

What you suggest it's not a bug but it's an improvement and a good one.
Thanks for the suggestion, the Product Team will take a look.
Please stay tuned !

Ping @PrestaShop/product-team , I agree with @juliemoynat that using the `for` and `id` combination for `label` and form fields is the more robust way to have accessible form fields, wdyt?

**Comment by NeOMakinG** [comment]
Hello @juliemoynat and finally, thanks for creating this issue :)

I'm currently working on it and here is a list of which inputs doesn't have any accessible labels :

- Login form inputs could have a for + id combination instead of aria-label
- Register form inputs could have a for + id combination and some are even missing aria-labels such as password and checkboxes
- Checkout forms got aria label instead of for + id combination
- Cart => Quantity input of every products doesn't have an aria-label attribute and doesn't have a label at all, I think about adding an aria-label attribute on this one 
- Shipping method looks unclear for a screen reader and I think an aria-label could fix it as the current markup can't be clearer without introducting breaking changes :/
- Contact form doesn't have any aria-label and no for + id

Also, the #23353 introduced aria-labels inside almost every input fields, so it's not aiming only 2 fields, but I feel like it could be clearer anyway :)

I think forms should be fixable globally, I'll work on it today, if you find some more missing accessibility labels, don't hesitate!

**Comment by jeckyl** [comment]
What is the justification for not using input inclusion or code for="" ?

The optimized Classic Rocket template integrates the use of for=" ".

https://github.com/prestarocket-agence/classic-rocket/blob/master/templates/_partials/form-fields.tpl

**Comment by NeOMakinG** [comment]
Where did you see that we'll not use for + id combination ? I wrote that we'll implement it !

**Comment by jeckyl** [comment]
> Where did you see that we'll not use for + id combination ? I wrote that we'll implement it !

> Also, the #23353 introduced aria-labels inside almost every input fields, so it's not aiming only 2 fields, but I feel like it could be clearer anyway :)
![Screenshot_2021-03-02 Fix classic theme images width, accessibilty and some lighthouse improvements by NeOMakinG · Pull Req](https://user-images.githubusercontent.com/595633/109631842-aae51880-7b46-11eb-8f74-b65cf9ede298.png)

I'm just reading the code you're giving us 

Sorry I didn't understand that this will also be fixed next time.

**Comment by NeOMakinG** [comment]
@jeckyl Yes that a previous PR, I'll modify it ;)

**Comment by hibatallahAouadni** [comment]
@NeOMakinG so shall we change the issue label to **`To Do`** or do we need the PM approval?

**Comment by NeOMakinG** [comment]
@hibatallahAouadni I don't think PM approval is needed as it doesn't change any visible behavior and shouldn't break anything, we can put the To Do label!

**Comment by Progi1984** [comment]
Fixed by @NeOMakinG in #23480
